packages/smooth_app/lib/pages/onboarding/country_selector.dart

```diff
@@ -0,0 +1,155 @@
+import 'package:iso_countries/iso_countries.dart';
+import 'package:openfoodfacts/utils/CountryHelper.dart';
+import 'package:smooth_ui_library/util/ui_helpers.dart';
+
+/// A selector for selecting user's country.
+class CountrySelector extends StatefulWidget {
+  const CountrySelector();
+
+  @override
+  State<CountrySelector> createState() => _CountrySelectorState();
+}
+
+class _CountrySelectorState extends State<CountrySelector> {
+  late List<Country> _countryList = <Country>[];
+  String? _chosenValue;
+  late Future<void> _initFuture;
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    _initFuture = _init();
+  }
+
+  Future<void> _init() async {
+    final String locale = Localizations.localeOf(context).languageCode;
+    final List<Country> localizedCountries =
+        await IsoCountries.iso_countries_for_locale(locale);
+    _countryList = sanitizeCountriesList(localizedCountries);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final AppLocalizations appLocalizations = AppLocalizations.of(context)!;
+    return FutureBuilder<void>(
+        future: _initFuture,
+        builder: (BuildContext context, AsyncSnapshot<void> snapshot) {
+          if (snapshot.hasError) {
+            return Text('Fatal Error: ${snapshot.error}');
+          }
+          if (snapshot.connectionState != ConnectionState.done) {
+            return const CircularProgressIndicator();
+          }
+          return Padding(
+            padding:
+                const EdgeInsets.only(top: MEDIUM_SPACE, bottom: LARGE_SPACE),
+            child: DropdownButtonFormField<String>(
+              value: _chosenValue,
+              style: const TextStyle(color: Colors.black),
+              decoration: InputDecoration(
+                enabledBorder: OutlineInputBorder(
+                  borderSide: const BorderSide(
+                      color: Color.fromARGB(255, 235, 235, 235)),
+                  borderRadius: BorderRadius.circular(VERY_LARGE_SPACE),
+                ),
+                filled: true,
+                fillColor: const Color.fromARGB(255, 235, 235, 235),
+              ),
+              items:
+                  _countryList.map<DropdownMenuItem<String>>((Country country) {
+                return DropdownMenuItem<String>(
+                  value: country.name,
+                  child: Text(country.name),
+                );
+              }).toList(),
+              hint: Text(
+                appLocalizations.country_chooser_label,
+                style: const TextStyle(
+                    color: Colors.black,
+                    fontSize: 16,
+                    fontWeight: FontWeight.w600),
+              ),
+              onChanged: (String? value) {
+                setState(() {
+                  if (value != null) {
+                    // TODO(jasmeet): Store pref in _userPreferences
```

Comment:
Still to be done, but that doesn't look too complicated.

Reply:
Done